[BMalloc] Add macro check to BUSE_TZONE feature define

msaboff opened the pull request:
#### 71b01c1ef9bad8f3e02baa0565e149552ef60e16
<pre>
[BMalloc] Add macro check to BUSE_TZONE feature define
<a href="https://bugs.webkit.org/show_bug.cgi?id=275078">https://bugs.webkit.org/show_bug.cgi?id=275078</a>
<a href="https://rdar.apple.com/129187840">rdar://129187840</a>

Reviewed by Ryan Haddad and Elliott Williams.

Added an &quot;#if !defined(BUSE_TZONE)&quot; check for the setting of BUSE_TZONE to allow the flag to be set with a compile time define.

* Source/bmalloc/bmalloc/BPlatform.h:

Canonical link: <a href="https://commits.webkit.org/279721@main">https://commits.webkit.org/279721@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7e78776115e80f9d509faeff8e1cd3748de2c892

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54154 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33541 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6690 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57430 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4878 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41060 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4774 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43856 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3258 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56250 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/31789 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46892 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25005 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28596 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4212 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3027 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/47513 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/50302 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4419 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59023 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/53662 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29355 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/4518 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51277 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30528 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47006 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50634 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11826 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31495 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/65963 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30309 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12559 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/bmalloc/bmalloc/BPlatform.h:
```diff
@@ -392,7 +392,7 @@
+#if !defined(BUSE_TZONE) && BUSE(LIBPAS) && BOS(DARWIN) && BCPU(ARM64)
 #define BUSE_TZONE 0
 #else
 #define BUSE_TZONE 0
```
Does this work?  It looks like if `defined(BUSE_TZONE)`, it will fall into the `#else` case which `#define BUSE_TZONE 0`

msaboff replied:
Fixed.

msaboff committed:
[BMalloc] Add macro check to BUSE_TZONE feature define
https://bugs.webkit.org/show_bug.cgi?id=275078
rdar://129187840

Reviewed by Ryan Haddad and Elliott Williams.

Added an "#if !defined(BUSE_TZONE)" check for the setting of BUSE_TZONE to allow the flag to be set with a compile time define.

* Source/bmalloc/bmalloc/BPlatform.h:

Canonical link: https://commits.webkit.org/279721@main
